code/modules/clothing/glasses/_glasses.dm

```diff
@@ -573,3 +573,38 @@
 	desc = "Lookin' cool."
 	icon_state = "phantom_glasses"
 	inhand_icon_state = "phantom_glasses"
+
+/obj/item/clothing/glasses/dealmaker
+	name = "dealmakers"
+	desc = "Why be the little sponge who hates its $4.99 life when you can be a BIG SHOT?"
+	icon_state = "dealmaker"
+	inhand_icon_state = "dealmaker"
+
+/obj/item/clothing/glasses/dealmaker/equipped(mob/M, slot)
+	. = ..()
+	if (slot == ITEM_SLOT_EYES)
+		RegisterSignal(M, COMSIG_MOB_SAY, .proc/handle_speech)
+	else
+		UnregisterSignal(M, COMSIG_MOB_SAY)
+
+/obj/item/clothing/glasses/dealmaker/dropped(mob/M)
+	. = ..()
+	UnregisterSignal(M, COMSIG_MOB_SAY)
+
+/obj/item/clothing/glasses/dealmaker/proc/handle_speech(datum/source, list/speech_args) //this should probably be changed at some point via some sort of refactor so that all clothing can modify speech
```

Comment:
this, and every other speech replacer, needs to be refactored into something better than duplicating procs a billion times over on different clothing items.

Reply:
My thought would be an element attached and detached to humans, that way it works with mutations that change text AND clothing that changes text